### ISSUE TITLE
Demo of Identify on Map Source type="ags"

### ISSUE THREAD

**Issue by tchaddad** [geomoose/gm3#730]
Fiddling with the demo of the `mapbook-test-servers.xml`, specifically the example of "ags-polygons" that uses this web service:
https://gis2.co.dakota.mn.us/arcgis/rest/services/DCGIS_OL_Transportation/MapServer/21

The server says the layer supports Query, and the Layer draws in the demo (3.9.1-beta) when turned on, but the Identify "Feature Info" seems to not work: instead it alerts no "feature to identify", even though clearly there are features. There are no errors reported.

I notice the service is in Spatial Reference: 26915, so I wondered if the identify is failing due to a projection issue? Or could it be a click tolerance issue?

The demo mapbook snippet has this for identify on that layer: `<template name="identify" auto="true" />`

Is that all that is required? Wondering if the mapbook-test-servers.xml is missing a parameter?

Demo is here: https://demo.geomoose.org/3.x/desktop/?mapbook=test

Screenshot of the specific Catalog entry, and feature location (Runway polygons):

![ags_identify](https://user-images.githubusercontent.com/1042618/166852186-b0c24882-b1d4-4fd2-b14e-f2c639fad784.JPG)

**Comment by brentfraser** [comment]
I ran across this while testing protocols (see #547).  Try the Identify with type="ags-vector"

**Comment by tchaddad** [comment]
Thanks @brentfraser I was able to get Feature Info to work using the template that is listed in the demo under "Vector with Feature Info", but the demo under "Image with Feature Info" did not work for me, and I tested with a couple of other servers in 3857 as well.. In fact, I did not even see the request get sent to the server, so maybe you can take a look to see if the demo is operating as you intended from the top level folder organization?

I'll circle back later today with a few more tests.

**Comment by brentfraser** [comment]
Based on #547, it never worked.  I guess we have to decide if it should (e.g file an Issue, enhance the code, add to doc).

**Comment by tchaddad** [comment]
Ah ok - I guess I was misinterpreting that table and how it relates to the Demo. I think it could point to a need for a Catalog label change on that layer as marked in the screenshot above, as currently it is in the group "Image with Feature Info" implying that it works, But I understand that Mapbook is for testing and so no harm. 

I will say that this does seem like a good candidate for a GM enhancement, as that service type is in wide use, and is more backward compatible for users on older Esri systems than anything based on Esri Feature Services.

**Comment by brentfraser** [comment]
Yeah the table and demo are out of sync.  We need to remove the "Image With Feature -> AGS" entries from the `mapbook-test-servers.xml` (or add some text indicating it is not supported).  And I need to move the Data Protocol support from the issue to the doc.

And it is a good candidate for enhancement.

**Comment by theduckylittle** [comment]
If folks have time before the next PSC, I'd love to review an updated Source/Feature support list. Could be an opportunistic refactor. :D

**Comment by brentfraser** [comment]
@theduckylittle , do you mean reviewing the supported protocols and identifying opportunities to improve Geomoose? e.g. review the tables in #547?

**Comment by tchaddad** [comment]
@brentfraser I think this might mean: are there any other items in the tables that GM does not support that we can highlight as enhancements. 

The tables are super useful for seeing all the capabilities of all the service types in one place, but there is no column or other set of notes that confirms missing support in GM aside from the one footnote '2'. 

Does that mean that footnote '2' is the only unsupported item in the table?

**Comment by brentfraser** [comment]
@tchaddad thanks for the question; it clarifies what @theduckylittle requesting.

The short answer: yes 

The long answer: yes, but the table is really only the tip of the iceberg.  It doesn't detail restrictions on projections, legends, query formats, returned data formats.  For insight into those see:

#475 Support for 'layers' param when map source type = ags
#517 pass thru additional parameters for AGS map-source
#524 WFS map-source and BBOX coordinates SRS
#542 Allow WFS query return to be in EPSG:4326
#546 Identify in WMS: allow GeoJSON as INFO_FORMAT
#556 ags-vector legend
#574 Support for external param value

**Comment by brentfraser** [comment]
TBD: Test with #775

**Comment by chughes-lincoln** [comment]
Not sure where things stand on this, but I was able to get identify working on one of my rest services using type ags-vector. More specifically, I set the layer to render as ags, but query-as ags-vector (similar to how the parcel layer is displayed as mapserver but queried as mapserver-wfs), as displaying as ags-vector was pretty slow for a more data-intensive layer.

**Comment by brentfraser** [comment]
The current version (3.12) supports TWO different mapsource configurations to make this work:
1. "ags" mapsource for display using query-as to reference an "ags-vector" mapsource for attribute queries (identify, etc)
2. "ags" mapsource for display using query-as to reference an "ags" mapsource (IF the server supports QUERY).

For an example have a look at the [mapbook-test-servers.xml](https://github.com/geomoose/gm3/blob/bbb23f09063a32c6908fff341e28dd6f8d4dfce0/examples/desktop/mapbook-test-servers.xml#L774).  It has a URL of
`https://gis2.co.dakota.mn.us/arcgis/rest/services/DCGIS_OL_Transportation/MapServer/export`
with (note the 28):
`<param name="layers" value="show:28" />`

The referenced query-as mapsource below it has a URL of
`https://gis2.co.dakota.mn.us/arcgis/rest/services/DCGIS_OL_Transportation/MapServer/28`
(note instead of "export" it has the layer number.
